src/generators.rs

```diff
@@ -2420,6 +2420,148 @@ pub fn barbell_graph(
+///
+/// :param int n: The number of nodes to generate the graph with.
+///
+/// :returns: The generated empty graph
+/// :rtype: PyGraph
+///
+/// .. jupyter-execute::
+///
+///  import rustworkx.generators
+///  from rustworkx.visualization import mpl_draw
+///
+///  graph = rustworkx.generators.empty_graph(5)
+///  mpl_draw(graph)
+///
+#[pyfunction(multigraph = true)]
+#[pyo3(text_signature = "(/, n, multigraph=True)")]
+pub fn empty_graph(py: Python, n: usize, multigraph: bool) -> PyResult<graph::PyGraph> {
+    let mut graph = StableUnGraph::<PyObject, PyObject>::default();
+    for _ in 0..n {
+        graph.add_node(py.None());
+    }
+    Ok(graph::PyGraph {
+        graph,
+        node_removed: false,
+        multigraph,
+        attrs: py.None(),
+    })
+}
+
+/// Generate a directed empty graph with ``n`` nodes and no edges.
+///
+/// :param int n: The number of nodes to generate the graph with.
+///
+/// :returns: The generated empty graph
+/// :rtype: PyDiGraph
+///
+/// .. jupyter-execute::
+///
+///  import rustworkx.generators
+///  from rustworkx.visualization import mpl_draw
+///
+///  graph = rustworkx.generators.directed_empty_graph(5)
+///  mpl_draw(graph)
+///
+#[pyfunction(multigraph = true)]
+#[pyo3(text_signature = "(/, n, multigraph=True)")]
+pub fn directed_empty_graph(
+    py: Python,
+    n: usize,
+    multigraph: bool,
+) -> PyResult<digraph::PyDiGraph> {
+    let mut graph = StableDiGraph::<PyObject, PyObject>::default();
+    for _ in 0..n {
+        graph.add_node(py.None());
+    }
+    Ok(digraph::PyDiGraph {
+        graph,
+        node_removed: false,
+        cycle_state: algo::DfsSpace::default(),
+        check_cycle: false,
+        multigraph,
+        attrs: py.None(),
+    })
+}
+
+/// Generate an undirected complete graph with ``n`` nodes.
+///
+/// A complete graph is a simple graph in which each pair of distinct
+/// vertices is connected by a unique edge.
+/// The complete graph on ``n`` nodes is the graph with the set of nodes
+/// ``{0, 1, ..., n-1}`` and the set of edges ``{(i, j) : i < j, 0 <= i < n, 0 <= j < n}``.
+/// The number of edges in the complete graph is ``n*(n-1)/2``.
+///
+/// :param int n: The number of nodes to generate the graph with.
+///
+/// :returns: The generated complete graph
+/// :rtype: PyGraph
+///
+/// .. jupyter-execute::
+///
+///  import rustworkx.generators
+///  from rustworkx.visualization import mpl_draw
+///
+///  graph = rustworkx.generators.complete_graph(5)
+///  mpl_draw(graph)
+///
+#[pyfunction(multigraph = true)]
+#[pyo3(text_signature = "(/, n, multigraph=True)")]
+pub fn complete_graph(py: Python, n: usize, multigraph: bool) -> PyResult<graph::PyGraph> {
+    let mut graph = empty_graph(py, n, multigraph)?;
+    if n > 1 {
+        for i in 0..n - 1 {
+            for j in i + 1..n {
+                graph.add_edge(i, j, py.None());
+            }
+        }
+    }
+    Ok(graph)
+}
```

Comment:
I think the complete graph function already exists, it's just called `mesh_graph`: https://github.com/Qiskit/rustworkx/blob/main/src/generators.rs#L553-L627. I think the complete graph name is probably more common. Maybe we should add this and just call `mesh_graph()` internally and document it as an alias.

Reply:
I don't think there is any real reference to the term, it was added in: https://github.com/Qiskit/rustworkx/pull/191 which was a long time ago. I didn't really think about the name at the time (especially since the library was relatively young then). I agree with you that `complete_graph` is a better name (this has come up a few times I think). For backwards compatibility we can't simply rename `mesh_graph` -> `complete_graph`. But for the first step just having `complete_graph` as an alias for `mesh_graph` would make it easier for people to fine the function they're looking for.

Reply:
I understand @mtreinish. I'll do that.